feat: make kubernetes owner parametrized

alegrey91 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

The following PR provide a new variable called `kube_owner`. It allows the user to set the kubernetes installation owner (default: `kube`).
This enable compliance with [CIS 1.1.19](https://www.tenable.com/audits/items/CIS_Kubernetes_v1.20_v1.0.0_Level_1_Master.audit:7c2372fabcc905e8ce10aed4ef02c853), setting the variable `kube_owner: root`.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #8948

**Special notes for your reviewer**:

This PR follows the job done in the previous months (#8773).

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Make kubernetes owner parametrized (using `kube_owner`/`kube_cert_group`/`etcd_owner` variables)
```

oomichi commented on roles/etcd/tasks/gen_certs_script.yml:
```diff
@@ -4,7 +4,7 @@
     group: "{{ etcd_cert_group }}"
     state: directory
-    owner: kube
+    owner: {{ kube_owner }}
```
Need to add " to pass a lint job as
```
The error appears to be in '/builds/kargo-ci/kubernetes-sigs-kubespray/roles/etcd/tasks/gen_certs_script.yml': line 7, column 13, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
    state: directory
    owner: {{ kube_owner }}
            ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:
    with_items:
      - {{ foo }}
Should be written as:
    with_items:
      - "{{ foo }}"
```

oomichi reviewed (commented):
most part seems good for me, thanks for doing this.

oomichi commented:
Thanks for updating.

/ok-to-test
/approve

alegrey91 commented:
Hi @oomichi,
Please, wait before merging.
I think to have found a bug when you deploy with option `etcd_deployment_type: host`.
I'll send the fix asap.

cristicalin commented:
@alegrey91 you can put the PR in draft until it is ready to prevent accidentally merging it

alegrey91 commented:
@oomichi, @cristicalin I think there's a bug in the installation procedure using the option `etcd_deployment_type: host`.
I did some tests, but this doesn't seem to be related to my PR. I tried to use also `kubespray` updated to the last commit (so, without my code) and I still have the same error (Ansible get stuck during task `"Gen_certs | Gather etcd member and admin certs from first etcd node"` at `roles/etcd/tasks/gen_certs_script.yml`). Maybe it's a permission issue.

alegrey91 commented:
> > @oomichi, @cristicalin I think there's a bug in the installation procedure using the option `etcd_deployment_type: host`.
> > I did some tests, but this doesn't seem to be related to my PR. I tried to use also `kubespray` updated to the last commit (so, without my code) and I still have the same error (Ansible get stuck during task `"Gen_certs | Gather etcd member and admin certs from first etcd node"` at `roles/etcd/tasks/gen_certs_script.yml`). Maybe it's a permission issue.
> 
> Can you collect some more details? Our default is `etcd_deployment_type: host` and we have not seen this issue reported before.

I was wrong. With the latest version the installation works fine.
Anyway, I'm still not able to understand what is happening with my fork.
I checked my code but I cannot undestand what could be the problem.
That's the configuration that I'm using for tests:
* `hardening.yaml`
```yaml
# Hardening
---
# this allow us to use PodSecurity admission plugin
# which is a built-in
kube_version: v1.23.6

## kube-apiserver
authorization_modes: ['Node','RBAC']
kube_apiserver_feature_gates: ['AppArmor=true']
kube_apiserver_request_timeout: 120s
kube_apiserver_service_account_lookup: true

# enable kubernetes audit
kubernetes_audit: true
audit_log_path: "/var/log/kube-apiserver-log.json"
audit_log_maxage: 30
audit_log_maxbackups: 10
audit_log_maxsize: 100

tls_min_version: VersionTLS12
tls_cipher_suites:
  - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
  - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
  - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305

# enable encryption at rest
kube_encrypt_secret_data: true
kube_encryption_resources: [secrets]
kube_encryption_algorithm: "secretbox"

kube_apiserver_enable_admission_plugins: ['EventRateLimit,AlwaysPullImages,SecurityContextDeny,ServiceAccount,NamespaceLifecycle,NodeRestriction,LimitRanger,ResourceQuota,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,PodNodeSelector,PodSecurity']
kube_apiserver_admission_control_config_file: true
# EventRateLimit plugin configuration
kube_apiserver_admission_event_rate_limits:
  limit_1:
    type: Namespace
    qps: 50
    burst: 100
    cache_size: 2000
  limit_2:
    type: User
    qps: 50
    burst: 100

kube_profiling: false
kube_proxy_profiling: false

## kube-controller-manager
kube_controller_manager_bind_address: 127.0.0.1
kube_controller_terminated_pod_gc_threshold: 50
kube_controller_feature_gates: ["RotateKubeletServerCertificate=true","AppArmor=true"]

## kube-scheduler
kube_scheduler_bind_address: 127.0.0.1
kube_kubeadm_scheduler_extra_args:
  profiling: false
kube_scheduler_feature_gates: ["AppArmor=true"]

## etcd
etcd_deployment_type: host

## kubelet
kubelet_authorization_mode_webhook: true
kubelet_authentication_token_webhook: true
kube_read_only_port: 0
kubelet_rotate_server_certificates: true
kubelet_protect_kernel_defaults: "true"
kubelet_event_record_qps: 1
kubelet_rotate_certificates: "true"
kubelet_streaming_connection_idle_timeout: "5m"
kubelet_make_iptables_util_chains: true
kubelet_feature_gates: ["RotateKubeletServerCertificate=true"]
```
* vars.yaml
```yaml
---

# K8s Cluster
kube_service_addresses: 10.96.0.0/12
kube_pods_subnet: 172.16.0.0/16
kube_proxy_mode: iptables
cluster_name: cke.labs
kubeconfig_localhost: true
kubectl_localhost: true
sans_lb_ip:
  - 192.168.20.2

# All
loadbalancer_apiserver_localhost: true
loadbalancer_apiserver_type: haproxy
upstream_dns_servers:
  - 8.8.8.8
  - 8.8.4.4
deploy_container_engine: true

# other
enable_nodelocaldns: false
```
* The command that I'm using is this
```bash
ansible-playbook -vvvvv ~/test/kubespray-test/cluster.yml \
        -i inventory.ini \
        -b --become-user=root \
        --private-key ~/.ssh/id_ecdsa \
        -e "@vars.yaml" \
        -e "@hardening.yaml"
```
* logs
```txt
<192.168.20.111> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/home/test/.ssh/id_ecdsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publi
ckey -o PasswordAuthentication=no -o 'User="test"' -o ConnectTimeout=10 -o 'ControlPath="/home/test/.ansible/cp/50a8eb98f0"' -tt 192.168.20.111 '/bin/sh -c '"'"'sudo -H -S -n  -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-
kbroxyopasrtkwbuwjyxcqvprnmxkmxy ; ALL_PROXY='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' FTP_PROXY='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'
"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' HTTPS_PROXY='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' HTTP_PROXY='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"
'"'"'"'"'"'"'"'"'"'"'"'"'"'"' NO_PROXY='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' all_proxy='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'
"'"'"'"'"'"'"'"'"'"'"'"' ftp_proxy='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' http_proxy='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"
'"'"'"'"'"'"'"'"'"'"' https_proxy='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"' no_proxy='"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"''"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'"'
"'"'"'"'"'"'"'"'"' /usr/bin/python3 /home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-227259892826231/AnsiballZ_copy.py'"'"'"'"'"'"'"'"' && sleep 0'"'"''                                                                       
Escalation succeeded                                                                                                                                                                                                                          
<192.168.20.111> (0, b'\r\n{"dest": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem", "src": "/home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-227259892826231/source", "md5sum": "5e2a8a48b2039e221cf0cfa7ad4925ae", "checksum":
 "49961d31ba01313489fc64e5667071042844e1c8", "changed": true, "uid": 997, "gid": 0, "owner": "kube", "group": "root", "mode": "0640", "state": "file", "size": 1679, "invocation": {"module_args": {"dest": "/etc/ssl/etcd/ssl/node-cke-worker
-01-key.pem", "group": "root", "owner": "kube", "mode": 416, "src": "/home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-227259892826231/source", "_original_basename": "tmp1uyh8i2q", "follow": false, "checksum": "49961d31ba013
13489fc64e5667071042844e1c8", "backup": false, "force": true, "unsafe_writes": false, "content": null, "validate": null, "directory_mode": null, "remote_src": null, "local_follow": null, "seuser": null, "serole": null, "selevel": null, "s
etype": null, "attributes": null}}}\r\n', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.4, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.con
f matched no files\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname 192.168.20.111 is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2
: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_reque
st_alive: done pid = 1969788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 192.168.20.1
11 closed.\r\n')                                                                                                                                                                                                                              
<192.168.20.111> ESTABLISH SSH CONNECTION FOR USER: test                                                                                                                                                                                   
<192.168.20.111> SSH: ansible.cfg set ssh_args: (-C)(-o)(ControlMaster=auto)(-o)(ControlPersist=60s)                                                                                                                                          
<192.168.20.111> SSH: ANSIBLE_PRIVATE_KEY_FILE/private_key_file/ansible_ssh_private_key_file set: (-o)(IdentityFile="/home/test/.ssh/id_ecdsa")                                                                                            
<192.168.20.111> SSH: ansible_password/ansible_ssh_password not set: (-o)(KbdInteractiveAuthentication=no)(-o)(PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey)(-o)(PasswordAuthentication=no)
<192.168.20.111> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User="test")                                                                                                                                          
<192.168.20.111> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=10)                                             
<192.168.20.111> SSH: Set ssh_common_args: ()              
<192.168.20.111> SSH: Set ssh_extra_args: ()               
<192.168.20.111> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath="/home/test/.ansible/cp/50a8eb98f0")                                                                                                                  
<192.168.20.111> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/home/test/.ssh/id_ecdsa"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publi
ckey -o PasswordAuthentication=no -o 'User="test"' -o ConnectTimeout=10 -o 'ControlPath="/home/test/.ansible/cp/50a8eb98f0"' 192.168.20.111 '/bin/sh -c '"'"'rm -f -r /home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-22
7259892826231/ > /dev/null 2>&1 && sleep 0'"'"''           
<192.168.20.111> (0, b'', b'OpenSSH_8.2p1 Ubuntu-4ubuntu0.4, OpenSSL 1.1.1f  31 Mar 2020\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no fil
es\r\ndebug1: /etc/ssh/ssh_config line 21: Applying options for *\r\ndebug2: resolve_canonicalize: hostname 192.168.20.111 is address\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hel
lo_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done p
id = 1969788\r\ndebug3: mux_client_request_session: session request sent\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
changed: [cke-controlplane-01] => (item=/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem) => {                             
    "ansible_loop_var": "item",                            
    "changed": true,                                       
    "checksum": "49961d31ba01313489fc64e5667071042844e1c8",                                                            
    "dest": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem",                                                            
    "diff": [],                                            
    "gid": 0,                                              
    "group": "root",                                       
    "invocation": {                                        
        "module_args": {                                   
            "_original_basename": "tmp1uyh8i2q",                                                                       
            "attributes": null,                            
            "backup": false,                               
            "checksum": "49961d31ba01313489fc64e5667071042844e1c8",                                                    
            "content": null,                               
            "dest": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem",                                                    
            "directory_mode": null,                        
            "follow": false,                               
            "force": true,                                 
            "group": "root",                               
            "local_follow": null,                          
            "mode": 416,                                   
            "owner": "kube",                               
            "remote_src": null,                            
            "selevel": null,                               
            "serole": null,                                
            "setype": null,                                
            "seuser": null,                                
            "src": "/home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-227259892826231/source",                                                                                                                                  
            "unsafe_writes": false,                        
            "validate": null                               
        }                                                  
    },                                                     
    "item": {                                              
        "ansible_loop_var": "item",                        
        "changed": false,                                  
        "content": "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcEFJQkFBS0NBUUVBbVJuMHRkRTZaRG9iOHhhZmN0WHl4Tzl0ZW1ZaGFOWCsrdHo3RTUzL2laYXlEblZCCjVRWXNYeHU2S3B2aXA2NkEyNy8wWTdnQkJpS3kwaTV6RUlQUThaR2FKRkRSaUNMRjg1SEpLRWhYWlRhcWdSMm4Kaj
JDMjhmRnVpUGZmbkROTWc5MDFpcHZmUVl5V2hCcEp1YWpBeERTcUN4aFZKRTBibURDbmxwYWh5cU50cGNqSAo0V2pKTjc3SDNFQWQzcXdxdTRHQXU3Yi92bjBhSzZFRmxTdW5aalZHMDRtN1ZLeHN3TzB5cDUwR3NyMWxjK0RWCkl4VHRrWVNNMjNoWVBrSG1hV1NTeVhJaC9lYXFqbWNnakladHZ3Q2toT0hVRkpTdzFQ
dkZteVJsd3dZc0pxZEQKWUhGRkVXRE01S2UzUVhpQldNMlQ0V1JSV3lRV2ozbm1UeTdhNFFJREFRQUJBb0lCQUJscGc0TENUY3poRUU2VApETGhFNXR6bFdPaWVGeUdOazVzVUh1Q3Nzd09sYmJ5VW5mRmp5dTJtYm9vd0Fia1BiL2dKcm9qMnU4VEc4YU9OCmJpL3BrVDduaWd4SkZkNWcxemFNbGpVSGZPb2gzcjg3cX
RwVGxVb25aczk1UlRacUZXTTF6bk5KV0E5TFJKaDgKdkl3Q2hLeU1pVVQ2THZldGpNMEFGbkVhWTVSOHRqY1EvQnFCQlRFRG5taURUeTlXSld6QnROWmdvQ3RaREFWYgpVTWVvT0JUYlVwYVEwUDFUZVFRd3Bnd2VuWTM4cjNWT0R3ZFNHeEw3cXpZNy9LYzBRV05VbUNlYUpUNHU2cmJlCnU2QklrYzd4MzU4UUhvMStj
WGlxTk5CTEpGZ3c5Mm55Sk5NdW9xaiswY2lRdFZrQUQvTy9YdkhEakpJWFQxUGQKTlN1TVVvRUNnWUVBeDdTY25vcHpYSm5WZzJSTGdmSnpPNmlUR0NpNVFxTmxFSXlUaUxxMThKKzZJMnNEc1cxKwplcGlHSjMrWHNobnNoVHN0amJNY3M0VjM0Ny9CTE1oS0ZLMXBWdWFocmw3SFVrOWRreXoyRk93dTlsY1lIb1RSCl
BtNkVaNWxSc2tXcWlLYWxRUENVS0FQU3Q1T3d3eWhmbit6SmVrWFZIWnJhWitZTnFETGZVM2tDZ1lFQXhFSTkKMU5HTEF3c3V6TG5QUHA2cndyY1RkY3hVL1NwUXJMTmF1T041Si9LQXpLcGlqMjRjYmNQVHc0eHlmTk5RclZkKwphaGY1VE0wbVd0WmJYZ2ZDL3VHNjI1VFEveDh3ZE1ZdVdPcU1yRzFCUTVpWEV0dkJC
aEpwVG1xQWphbkNRVXU1CjdscmlZNFdLMFRsZlRkelJVTW0rU0YyemU3bkxtaWpxZ0JUV3dLa0NnWUVBeDN4eUNxaHdUaXJZcmxRc1hOVU0KS0tiZXNINVZlT3h3MHhRcGQ2WlNnZGQ3LzA4eHREV1U1dStlNnF1NG8wN2tCYjQwWmR4cUtpV0R1ZnZkQWxmYwpNRVY4K1RLT2hQam1FQ0Y1YnIrMTNQYUVmc3hIS28wdj
J6Wllpb2dMcEpHVXBNWWFRd0ZPYlo1TUNkYTg5bVZKCkNCNW1yLzQ3d1VQak1YQUxhbDNPMmJrQ2dZQnk3OHhnVDV1VzI3eTd2dkhsWFcxLzZsSG1RaFZmOHFFdWt4ZEYKZ2txWkg0ZFkxVXhGYlpvT21vbnhVdFpWempmSVhKNDNLZlVoaWwrMUFTYWZLZktEN05UTzloZHo0ZU14OEVLQwpabEtlUkpuZEtZY0lUMVI2
NXR0dlJpWWxVTTdvQ0o1S1BrV3BQVE13RnVaK3BoeWJzSnJVckxPSmVTVUpFbjlMCkpXa2ppUUtCZ1FDZ3RwNmdCcHZDYnB3TUZkdGJHcENmQ0ZYRCtDOUFVMWhHWVQvSVNTUnBhMzhKa215NzdTVUYKdlFaTzFMaS90N2dSVzAyd0pjaHdvbVRnU0lkOGhSVmswK0cwUXllNG9nNit2UXJBM04zdFFSRnBuUklRWDJEbw
prV1dQWUliLzhKQVpQeUIvaGExWjd1R0tNUWlHaGp4MERqQ1lOa0QzYVlDREhaLytTT2hoV3c9PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=",                                                                                                                      
        "encoding": "base64",
        "failed": false,                                   
        "invocation": {                                    
            "module_args": {                               
                "src": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem"                                                  
            }                                              
        },                                                 
        "item": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem",                                                        
        "source": "/etc/ssl/etcd/ssl/node-cke-worker-01-key.pem"                                                       
    },                                                     
    "md5sum": "5e2a8a48b2039e221cf0cfa7ad4925ae",                                                                      
    "mode": "0640",                                        
    "owner": "kube",                                       
    "size": 1679,                                          
    "src": "/home/test/.ansible/tmp/ansible-tmp-1655218294.70278-1989496-227259892826231/source",                                                                                                                                          
    "state": "file",                                       
    "uid": 997                                             
}
```
Ansible get stuck at this point without terminating or giving some error.
If I change the configuration to `etcd_deployment_type: kubeadm` it works peoperly.

alegrey91 commented:
The last commit should solve the previous issue. It was related to wrong file permissions under `/etc/ssl/etcd/ssl/` (cannot be owned by `root`).
I also added a task (only for **etcd** installed as a container), that set the ownership of `/var/lib/etcd/` to `etcd:etcd`, in order to pass the [CIS 1.1.12](https://www.tenable.com/audits/items/CIS_Kubernetes_v1.20_v1.0.0_Level_1_Master.audit:1393dcc2ecf1b359f29d93d1cc16f083).

oomichi commented on roles/etcd/tasks/gen_certs_script.yml:
```diff
@@ -4,7 +4,7 @@
     group: "{{ etcd_cert_group }}"
     state: directory
-    owner: kube
+    owner: "{{ etcd_owner }}"
```
Could you explain why the default etcd owner is changed from `kube` to `etcd`?

alegrey91 replied:
Yes. In this PR I introduced the variable `kube_owner`. Its default value is `kube`, but if you want set a different user like `root`, then `kube` will never be created. Also (as I mentioned in the comment above), I noticed that setting the user `root` to manage `etcd` files, led to an installation issue. So, for these reasons, I thought to set the `etcd` user (that is already created by kubespray) for the `etcd` files.

oomichi replied:
Thank you for your explanation, I got it.

oomichi commented on roles/kubernetes/control-plane/tasks/kubeadm-etcd.yml:
```diff
@@ -16,3 +16,10 @@
+- name: Set ownership for etcd data directory
+  file:
+    path: "{{ etcd_data_dir }}"
+    owner: etcd
```
Don't we need to specify "{{ etcd_owner }}" instead of a static `etcd`?

alegrey91 replied:
Yes, probably is better.

oomichi replied:
Sorry for that I could not get the point, which is better?

alegrey91 replied:
Sorry for misunderstanding. I meant, probably is metter to add the variable `{{ etcd_owner }}` instead of the static `etcd`.

oomichi replied:
ah, OK.
Then I will add approval again after updating the above place.

oomichi reviewed (commented):
temporary approval removal for understanding this change correctly.

/approve cancel

oomichi commented:
Thanks for updating again.

/approve

floryut reviewed (approved):
@alegrey91 Thank you

/lgtm
